bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/LaunchConfigurationDelegate.java

```diff
@@ -446,8 +448,22 @@ else if (sessionType == SessionType.CORE)
 		}
 		catch (ExecutionException e1)
 		{
-			throw new DebugException(new Status(IStatus.ERROR, GdbPlugin.PLUGIN_ID, DebugException.REQUEST_FAILED,
-					"Error in services launch sequence", e1.getCause())); //$NON-NLS-1$
+			if (e1.getMessage().contains("Starting OpenOCD timed out.")) //$NON-NLS-1$
+			{
+				Display.getDefault().asyncExec(() -> {
+					ServerTimeoutErrorDialog.openError(Display.getDefault().getActiveShell());
+
```

Comment:
See if you can propogate back this exception with appropriate message to the UI class and handle there

Reply:
propagated via statusHandler in the latest commit